manager/4337/README.md

```diff
@@ -0,0 +1,114 @@
+# Safe{Core} Protocol ERC-4337 Support
+
+## Compatibility
+
+Safe{Core} Protocol specification is meant to be compatible with [ERC-4337](https://eips.ethereum.org/EIPS/eip-4337). ERC-4337 enforces rules on the the read/write operations that should be carefully looked into for the Safe{Core} Protocol implementation to be compatible with ERC-4337.
+
+As per ERC-4337 specs state the following when simulating a `UserOperation`:
+
+> Storage access is limited as follows:
+>
+> - self storage (of factory/paymaster, respectively) is allowed, but only if self entity is staked
+> - account storage access is allowed (see Storage access by Slots, below),
+> - in any case, may not use storage used by another UserOp sender in the same bundle (that is, paymaster and factory are not allowed as senders
+
+Storage associated with an account is defined as follows:
+
+> An address `A` is associated with:
+>
+> - Slots of contract `A` address itself.
+> - Slot `A` on any other address.
+> - Slots of type `keccak256(A || X) + n` on any other address. (to cover `mapping(address => value)`, which is usually used for balance in ERC-20 tokens). `n` is an offset value up to 128, to allow accessing fields in the format `mapping(address => struct)`
+
+As such, in order for the Safe{Core} Protocol to be ERC-4337 compatible without requiring a staked _paymaster_, the Manager MUST NOT make use of storage slots not associated with the account whose `UserOperation`s it is validating. Notably, this implies that `UserOperation` validation MUST be implemented without interacting with the Registry. If ERC-4337 `validateUserOp` support is implemented as a module, the Manager MUST NOT verify it is approved in the Registry. Note that this restriction only applies to `UserOperation` validation flow and notably does not apply to the `UserOperation` execution flow. The specification currently suggests that ERC-4337 `validateUserOp` SHOULD be directly supported in the Manager for these compatability reasons.
+
+Additionally, developers aiming to develop plugins that are ERC-4337 compatible should be aware of storage access restrictions, opcode usage restrictions during the simulation step of ERC-4337 specification.
+
+More details on this is available [here](https://github.com/safe-global/safe-core-protocol/issues/60#issuecomment-1761296305).
+
+### Staking
+
+The account SHOULD implement a staked _factory_ in order to enable `UserOperations` with `initCode`. ERC-4337 requires a staked _factory_ for `UserOperation` validation that reads associated account storage when `initCode` is specified. Since this specification permits Manager implementations that read account associated storage during `UserOperation` validation, the account _factory_ needs to be staked in order for bundlers to accept `UserOperations` with `initCode`.
```

Comment:
Could we add more context what storage is accessed that make this necessary? I.e. for `validateUserOp` it is outlined that the registry check is problematic. What would be problematic here?